ReactiveCocoaTests/Swift/SignalSpec.swift

```diff
@@ -453,6 +473,33 @@ class SignalSpec: QuickSpec {
 				observer.sendNext(1)
 				expect(lastValue).to(equal("2"))
 			}
+
+			it("should not keep resulting signal alive indefinitely") {
+				weak var signal: Signal<AnyObject, NoError>? = Signal.never.map { $0 }
+
+				expect(signal).to(beNil())
+			}
+
+			it("should not keep resulting signal alive indefinitely even if it has an observer") {
```

Comment:
> In fact, this test case succeeds

Sorry, that's "should not keep signal alive indefinitely even if it has an observer" that passes, the one without map (which should not make any difference). Anyway, if signal is to be alive until it terminates (even if can never terminate) then "should not keep signal alive indefinitely" (that had been there for a long time) should also fail. The only way for it not to fail is to have a retain cycle inside `Signal.never` itself, which is ridiculous. Btw deallocation of a signal here does not affect (stop) any underlying work (in case it's still being performed, but lost its reference to the generator observer and is not going to send events anymore) because it does not call the generator disposable (which could stop the work) as it would if signal terminated.